virtualcluster/pkg/controller/controller.go

```diff
@@ -37,6 +37,19 @@ func (c *Controllers) SetupWithManager(mgr ctrl.Manager) error {
 	opts := controller.Options{
 		MaxConcurrentReconciles: c.MaxConcurrentReconciles,
 	}
+
+	// If Provisioner is CAPI exit fast and only implement CAPI
+	// VC reconciler.
+	if c.ProvisionerName == "capi" {
+		if err := (&controllers.ReconcileCAPIVirtualCluster{
+			Client: mgr.GetClient(),
+			Log:    c.Log.WithName("virtualcluster"),
```

Comment:
We're still reconciling `virtualcluster` resources so I don't think so and they can't be turned on with other reconcilers since it's done via the `--master-prov` flag. Also we don't do this with `native` vs `aliyun`:
* https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/virtualcluster/pkg/controller/controller.go#L52